.github/workflows/linuxtests.yml

```diff
@@ -0,0 +1,58 @@
+name: Linux Tests
+
+on:
+  push:
+    branches:
+      - master
+      - maintenance/**
+  pull_request:
+    branches:
+      - master
+      - maintenance/**
+
+jobs:
+  Python_3.7_Debug
+    name: Python 3.7 Debug
+    runs-on: ubuntu-18.04
+    env:
+      USE_DEBUG=python3.7-dbg
+      TESTMODE=fast
+      NUMPYSPEC="--upgrade numpy"
+    steps:
+    - uses: actions/checkout@v2
+    - name: Create a venv for python-dbg runs-on
+      run: |
+        virtualenv --python=$USE_DEBUG venv
+        source venv/bin/activate
+        apt install python3.7-dbg python3.7-dev libatlas-base-dev liblapack-dev gfortran libgmp-dev libmpfr-dev libsuitesparse-dev ccache swig libmpc-dev
+    - name: Configure env vars and python packages
+      run: |
+        free -m
+        export PATH=/usr/lib/ccache:$PATH
+        ccache --zero-stats
+        export CCACHE_COMPRESS=1
```

Comment:
I would try getting it working without the ccache to start with. This requires quite a bit of fiddling around with. See https://cristianadam.eu/20200113/speeding-up-c-plus-plus-github-actions-using-ccache/ for more details.